contracts/interchain-token-service/src/contract/execute/mod.rs

```diff
@@ -390,13 +390,21 @@ fn register_chain(storage: &mut dyn Storage, config: msg::ChainConfig) -> Result
     }
 }
 
-pub fn update_chain(
-    deps: DepsMut,
-    chain: ChainNameRaw,
-    its_address: Address,
-) -> Result<Response, Error> {
-    state::update_its_contract(deps.storage, &chain, its_address).change_context(Error::State)?;
-    Ok(Response::new())
+pub fn update_chains(deps: DepsMut, chains: Vec<msg::ChainConfig>) -> Result<Response, Error> {
+    chains
+        .into_iter()
+        .map(|chain_config| update_chain(deps.storage, chain_config))
+        .try_collect::<_, Vec<Response>, _>()?
+        .then(|_| Ok(Response::new()))
```

Comment:
use try_for_each instead. You're not really interested in the returned object unless it's an error